vault/identity_store_oidc.go

```diff
@@ -0,0 +1,743 @@
+package vault
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"fmt"
+	"time"
+
+	"gopkg.in/square/go-jose.v2"
+
+	uuid "github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/helper/parseutil"
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+// types
+type audience []string
+type signingAlgorithm int
+
+const (
+	rs256 signingAlgorithm = iota
+)
+
+// globals - todo fix this
+var publicKeys []ExpireableKey = make([]ExpireableKey, 0, 0)
+
+type ExpireableKey struct {
+	Key       jose.JSONWebKey `json:"key"`
+	Expirable bool            `json:"expirable"`
+	ExpireAt  time.Time       `json:"expire_at"`
+}
+
+type NamedKey struct {
+	Name             string           `json:"name"`
+	SigningAlgorithm signingAlgorithm `json:"signing_algorithm"`
+	Verificationttl  string           `json:"verification_ttl"`
```

Comment:
```suggestion
	VerificationTTL  string           `json:"verification_ttl"`
```

Reply:
I'd expect this to be an integer duration, similar to our handling of other TTLs.  Same comment for RotationPeriod.